### PR TITLE
Feature/zen 21383.2

### PR DESCRIPTION
**Pull request by jpeacock-zenoss**

### CODE REVIEW

File: Products/ZenUtils/controlplane/application.py [hunk 1]
```diff
@@ -13,20 +13,146 @@
...
+        """
+        since = int((time.time() - self._lastUpdate) * 1000)
+        # No refresh if no time has elapsed since the last update
+        if since == 0:
+            return
+        changes = self._client.getChangesSince(since)
+        if changes:
+            # Changes exist, so refresh the last update time.
+            self._lastUpdate = time.time()
+        for changed_svc in changes:
```
Comment (rDr4g0n) [inline Products/ZenUtils/controlplane/application.py]: should this for...in be hit even if there are no changes?

Comment (jpeacock-zenoss) [inline Products/ZenUtils/controlplane/application.py]: The `getChangesSince` call usually returns an empty list so the `for` won't execute any iterations.

Comment (jpeacock-zenoss) [inline Products/ZenUtils/controlplane/application.py]: The `getChangesSince` call itself appears to be pretty quick.

Comment (rDr4g0n) [inline Products/ZenUtils/controlplane/application.py]: Got it.

**Pull request merged**